src/api/spec/factories/bs_requests.rb

```diff
@@ -24,17 +31,42 @@
     end
 
     bs_request_actions do |evaluator|
-      [build(:bs_request_action,
-             type: evaluator.type,
-             source_project: evaluator.source_project,
-             source_package: evaluator.source_package,
-             source_rev: evaluator.source_rev,
-             target_project: evaluator.target_project,
-             target_package: evaluator.target_package,
-             target_repository: evaluator.target_repository,
-             target_releaseproject: evaluator.target_releaseproject,
-             role: evaluator.role,
-             person_name: evaluator.person_name)]
+      attribs = attributes_for(:bs_request_action,
+                               type: evaluator.type,
+                               source_project: evaluator.source_project,
+                               source_package: evaluator.source_package,
+                               source_rev: evaluator.source_rev,
+                               target_project: evaluator.target_project,
+                               target_package: evaluator.target_package,
+                               target_repository: evaluator.target_repository,
+                               target_releaseproject: evaluator.target_releaseproject,
+                               role: evaluator.role,
+                               group_name: evaluator.group_name,
+                               person_name: evaluator.person_name)
+
+      if attribs[:source_project].is_a?(Project)
+        attribs[:source_project] = attribs[:source_project].name
+      end
+
+      if attribs[:target_project].is_a?(Project)
+        attribs[:target_project] = attribs[:target_project].name
+      end
+
+      if attribs[:source_package].is_a?(Package)
+        attribs[:source_project] ||= attribs[:source_package].project.name
+        attribs[:source_package] = attribs[:source_package].name
+      end
+      if attribs[:target_package].is_a?(Package)
+        attribs[:target_project] ||= attribs[:target_package].project.name
+        attribs[:target_package] = attribs[:target_package].name
+      end
+      attribs[:type] = attribs[:type].to_s
```

Comment:
why the `to_s`? :thinking:

Reply:
you don't want to know - really :)

But long story short: because the controller sets it as string as parsed out of XML. And it's really interesting how it breaks if you pass a symbol, but I find the rspec code easier to read if passing a symbol.